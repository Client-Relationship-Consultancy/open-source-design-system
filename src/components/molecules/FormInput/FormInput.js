import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import Label from "../../atoms/Label"
import Input from "../../atoms/Input"

// Declare CSS values for Container
const Container = styled.div`
  margin-bottom: 1rem;
  display: flex;
  flex-direction: column-reverse;
`
Container.displayName = "Container"

// Declare FormInput React Component
const FormInput = props => {
  // Refactor values from props (ES6)
  const {
    className,
    name,
    value,
    id,
    type,
    onChange,
    placeholder,
    label,
    disabled,
    bgColour,
    border,
    icon,
  } = props
  return (
    <Container className={className}>
      <Input
        name={name}
        value={value}
        id={id}
        type={type}
        onChange={onChange}
        placeholder={placeholder}
        disabled={disabled}
        bgColor={bgColour}
        border={border}
        icons={icon}
      />
      <Label htmlFor={id}>{label}</Label>
    </Container>
  )
}

// Declare the display name of component for static build
// Display name should match export name
FormInput.displayName = "FormInput"

// Declare propTypes values for FormInput
FormInput.propTypes = {
  /** Display Text */
  label: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  /** Placeholder is not a replacement for label as it is not optimal for handicapped users */
  placeholder: PropTypes.string,
  /** Needed when making controlled inputs for React */
  onChange: PropTypes.func.isRequired,
  /** Needed when making controlled inputs for React */
  value: PropTypes.string.isRequired,
  className: PropTypes.string,
  bgColour: PropTypes.string,
  border: PropTypes.string,
  icon: PropTypes.string,
}

export default FormInput
