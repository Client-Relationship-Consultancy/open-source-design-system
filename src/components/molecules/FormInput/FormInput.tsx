import React from "react";
import styled from "styled-components";

import Label from "../../atoms/Label";
import Input, { IInputProps } from "../../atoms/Input";

const Container = styled.div`
  margin-bottom: 1rem;
  display: flex;
  flex-direction: column-reverse;
`;

Container.displayName = "Container";

interface IFormInputProps extends Omit<IInputProps, "color" | "required" | "onFocus"> {
  label?: string;
}

const FormInput: React.FC<IFormInputProps> = (props) => {
  const {
    className,
    type,
    onChange,
    placeholder,
    id,
    name,
    value,
    label,
    disabled,
    icon,
    bgColor,
    border,
  } = props;
  return (
    <Container className={className}>
      <Input
        name={name}
        value={value}
        id={id}
        type={type}
        onChange={onChange}
        placeholder={placeholder}
        disabled={disabled}
        bgColor={bgColor}
        border={border}
        icon={icon}
      />
      <Label htmlFor={id}>{label}</Label>
    </Container>
  );
};

FormInput.displayName = "FormInput";

export default FormInput;
