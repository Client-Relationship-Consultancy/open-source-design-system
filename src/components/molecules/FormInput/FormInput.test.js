import React from "react"
import { shallow } from "enzyme"
import "jest-styled-components"

import FormInput from "./FormInput"

describe("FormInput test", () => {
  it("single line input matches the last snapshot", () => {
    const component = shallow(
      <FormInput label="Label" name="firstName" type="text" value="Bob" onChange={() => {}} />,
    )
    expect(component).toMatchSnapshot()
  })
})
