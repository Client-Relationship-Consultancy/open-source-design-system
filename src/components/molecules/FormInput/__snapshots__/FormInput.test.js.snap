// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput test single line input matches the last snapshot 1`] = `
<Container>
  <Input
    name="firstName"
    onChange={[Function]}
    type="text"
    value="Bob"
  />
  <Label>
    Label
  </Label>
</Container>
`;
